entrypoint.sh

```diff
@@ -3,15 +3,74 @@ set -e
 
 ## ProxySQL entrypoint
 ## ===================
+PROXYSQL_CONF_CHECK_INTERVAL=${PROXYSQL_CONF_CHECK_INTERVAL:-60}
+PROXYSQL_CONF_LIVE_RELOAD=${PROXYSQL_CONF_LIVE_RELOAD:-false}
+PROXYSQL_CONF_DIR=/tmp/proxysql-conf
+PROXYSQL_CONF_TPL_FILE=/tmp/proxysql-conf/proxysql.cnf.tpl
+PROXYSQL_CONF_FILE=/etc/proxysql.cnf
+PROXYSQL_SECRETS_DIR=/tmp/proxysql-secrets
+PROXYSQL_SECRETS_FILE=/tmp/proxysql-secrets/secrets.env
+PROXYSQL_CONF_RELOADER_LOCK=/tmp/proxysql-config-reloader.lock
 
-# Render /etc/proxysql.cnf from proxysql.cnf template
-if [ -f /tmp/proxysql.cnf ]; then
-	cat /tmp/proxysql.cnf | envsubst > /etc/proxysql.cnf
-fi
+function loadEnv {
+    eval $(cat $PROXYSQL_SECRETS_FILE | sed 's/^/export /') && \
+    echo "Loaded ProxySQL env vars"
+}
 
-# If command has arguments, prepend proxysql
-if [ "${1:0:1}" = '-' ]; then
-	CMDARG="$@"
-fi
+function renderProxySQLCnf {
+    cat $PROXYSQL_CONF_TPL_FILE | envsubst > $PROXYSQL_CONF_FILE && \
+    echo "Rendered $PROXYSQL_CONF_TPL_FILE"
+}
 
-exec proxysql -f $CMDARG
+function loadProxySQLCnf {
+    MYSQL_CMD='mysql -h 127.0.0.1 -P 6032 -u ${proxysql_admin_username} -p${proxysql_admin_password} -D main -e "LOAD MYSQL VARIABLES FROM CONFIG; LOAD MYSQL SERVERS FROM CONFIG; LOAD MYSQL USERS FROM CONFIG; LOAD MYSQL QUERY RULES FROM CONFIG; LOAD MYSQL VARIABLES TO RUNTIME; LOAD MYSQL SERVERS TO RUNTIME; LOAD MYSQL USERS TO RUNTIME; LOAD MYSQL QUERY RULES TO RUNTIME; SAVE MYSQL VARIABLES TO DISK; SAVE MYSQL SERVERS TO DISK; SAVE MYSQL USERS TO DISK; SAVE MYSQL QUERY RULES TO DISK;"'
```

Comment:
Do the `proxysql_admin_username` & `proxysql_admin_password` values come from the same mounted secret volume as the rest of the config? If so, what would happen if we tried to update these values?

Reply:
Nice point @sl1pm4t . AFAIK, updating admin credentials will require the pods to be rotated.